@utahdts/utah-design-system/react/components/forms/TimeInput.jsx

```diff
@@ -0,0 +1,171 @@
+ * @param {import('react').Ref<HTMLDivElement>} [props.innerRef]
+ * @param {boolean} [props.isClearable] should the clearable "X" icon be shown; is auto set to true if onClear is passed in
+ * @param {boolean} [props.isDisabled]
+ * @param {boolean} [props.isRequired]
+ * @param {string} props.label
+ * @param {string} [props.labelClassName]
+ * @param {string} [props.name]
+ * @param {(newValue: string) => void} [props.onChange] can be omitted to be uncontrolled OR controlled by form
+ * @param {() => void} [props.onClear] (not needed if inside a <Form> context)
+ * @param {string} [props.placeholder]
+ * @param {string} [props.timeFormat] use `date-fns` modifiers for formatting the time options
+ * @param {number} [props.timeRangeIncrement] for popup, what increment (in minutes) for the options given to the user
+ * @param {string} [props.timeRangeBegin] options in popup can start (inclusive) at a given time; format per `props.timeFormat`
+ * @param {string} [props.timeRangeEnd] options in popup can end at the given time (inclusive); format per `props.timeFormat`
+ * @param {string} [props.value]
+ * @param {string} [props.wrapperClassName]
+ * @returns {import('react').JSX.Element}
+ */
+export function TimeInput({
+  allowCustomEntry,
+  className,
+  defaultValue,
+  errorMessage,
+  hasTimePopup = true,
+  id,
+  innerRef,
+  isClearable,
+  isDisabled,
+  isRequired,
+  label,
+  labelClassName,
+  name,
+  onChange,
+  onClear,
+  placeholder,
+  timeFormat = 'h:mm aaa',
+  timeRangeBegin,
+  timeRangeEnd,
+  timeRangeIncrement = 15,
+  value,
+  wrapperClassName,
+  ...rest
+}) {
+  const {
+    onChange: currentOnChange,
+    onClear: currentOnClear,
+    value: currentValue,
+  } = useFormContextInputValue({
+    defaultValue,
+    id,
+    onChange,
+    onClear,
+    value,
+  });
+
+  const timeOptions = useMemo(
+    () => {
+      const defaultStartDate = new Date(new Date().setHours(0, 0, 0, 0));
+      const defaultEndDate = new Date(new Date().setHours(23, 59, 0, 0));
+      const optionsBeginDate = defaultDate((timeRangeBegin && parse(timeRangeBegin, timeFormat, new Date())) || null, defaultStartDate);
+      const optionsEndDate = defaultDate((timeRangeEnd && parse(timeRangeEnd, timeFormat, new Date())) || null, defaultEndDate);
+
+      const timeOptionsRet = [];
+      for (
+        let loopDate = optionsBeginDate;
+        loopDate.getTime() <= optionsEndDate.getTime();
+        loopDate = add(loopDate, { minutes: timeRangeIncrement })
+      ) {
+        timeOptionsRet.push(format(loopDate, timeFormat));
+      }
+
+      return timeOptionsRet;
+    },
+    [timeRangeBegin, timeRangeEnd, timeRangeIncrement]
+  );
+
+  const clockIcon = useMemo(
+    () => (
+      <span className={joinClassNames('utds-icon-before-clock', isDisabled && 'time-input__clock-icon--is-disabled', !hasTimePopup && 'time-input__clock-icon--static')} aria-hidden="true" />
```

Comment:
Icon is used for the icon button (combobox) and for a static input. It makes it difficult to style them exactly the same way. I don't think anyone will notice.